addons/stock/models/stock_location.py

```diff
@@ -221,7 +240,7 @@ def _get_putaway_strategy(self, product, quantity=0, package=None, packaging=Non
 
         # get current product qty (qty in current quants and future qty on assigned ml) of all child locations
         qty_by_location = defaultdict(lambda: 0)
-        locations = self.env['stock.location'].search([('id', 'child_of', self.id), ('usage', '=', 'internal')])
+        locations = self.child_internal_location_ids
```

Comment:
You can use the new field in `_get_putaway_location` too.

Reply:
already done :wink: